core/promise/consumer.go

```diff
@@ -32,17 +53,60 @@ func (c *consumer) NewRequest() (requestPtr interface{}) {
 }
 
 func (c *consumer) Consume(requestPtr interface{}) (response interface{}, err error) {
-	// request := requestPtr.(*Request)
+	request, ok := requestPtr.(*Request)
+	if !ok {
+		return nil, errUnsupportedRequest
+	}
+
+	receivedPromise, err := json.Marshal(request.SignedPromise.Promise)
+	if err != nil {
+		return nil, err
+	}
+
+	signature := identity.SignatureBase64(string(request.SignedPromise.IssuerSignature))
+	issuer := identity.FromAddress(request.SignedPromise.Promise.IssuerID)
+	verify := identity.NewVerifierIdentity(issuer)
```

Comment:
this could be better named as verifier, since the line below "verify.Verify" does seem a bit confusing